### ISSUE TITLE
OOM when finetuning FLANT5-xxl

### ISSUE THREAD

**Issue by AndrewZhe** [philschmid/deep-learning-pytorch-huggingface#6]
I use 4 A40 48G to finetune a FLANT5-XXL, following your blog "https://www.philschmid.de/fine-tune-flan-t5-deepspeed#2-fine-tune-flan-t5-xxl-using-deepspeed".  However, I met OOM with batch_size = 1 per GPU. 

I use fp16 and with offload off. If I turn on the offload, I could run the code and each GPU uses 22G VM.

Do you have any suggestions?

**Comment by philschmid** [comment]
What deepspeed config did you use? T5 is not working `fp16` also.

**Comment by AndrewZhe** [comment]
Sorry, it is bf16.

I use [ds_flan_t5_z3_config_bf16.json](https://github.com/philschmid/deep-learning-pytorch-huggingface/blob/main/training/configs/ds_flan_t5_z3_config_bf16.json)

{
  "bf16": {
    "enabled": "auto"
  },
  "optimizer": {
    "type": "AdamW",
    "params": {
      "lr": "auto",
      "betas": "auto",
      "eps": "auto",
      "weight_decay": "auto"
    }
  },
  "scheduler": {
    "type": "WarmupLR",
    "params": {
      "warmup_min_lr": "auto",
      "warmup_max_lr": "auto",
      "warmup_num_steps": "auto"
    }
  },
  "zero_optimization": {
    "stage": 3,
    "overlap_comm": true,
    "contiguous_gradients": true,
    "sub_group_size": 1e9,
    "reduce_bucket_size": "auto",
    "stage3_prefetch_bucket_size": "auto",
    "stage3_param_persistence_threshold": "auto",
    "stage3_max_live_parameters": 1e9,
    "stage3_max_reuse_distance": 1e9,
    "stage3_gather_16bit_weights_on_model_save": false
  },
  "gradient_accumulation_steps": "auto",
  "gradient_clipping": "auto",
  "steps_per_print": 2000,
  "train_batch_size": "auto",
  "train_micro_batch_size_per_gpu": "auto",
  "wall_clock_breakdown": false
}

**Comment by AndrewZhe** [comment]
Thanks! I will try to find another 4 gpus and try it.

 It is really attractive that 8 * A100 40G could use batch_size = 8 per GPU, without offloading.
